clients/pkg/promtail/client/client.go

```diff
@@ -318,23 +333,20 @@ func (c *client) sendBatch(tenantID string, batch *batch) {
 					level.Warn(c.logger).Log("msg", "error converting stream label string to label.Labels, cannot update lagging metric", "error", err)
 					return
 				}
-				var lblSet model.LabelSet
+				lblSet := make(prometheus.Labels)
 				for i := range lbls {
-					for _, lbl := range c.cfg.StreamLagLabels {
+					for _, lbl := range c.streamLagLabels {
 						if lbls[i].Name == lbl {
-							if lblSet == nil {
-								lblSet = model.LabelSet{}
-							}
-
-							lblSet = lblSet.Merge(model.LabelSet{
-								model.LabelName(lbl): model.LabelValue(lbls[i].Value),
-							})
+							lblSet[lbl] = lbls[i].Value
 						}
 					}
 				}
 				if lblSet != nil {
 					// always set host
-					lblSet = lblSet.Merge(model.LabelSet{model.LabelName(HostLabel): model.LabelValue(c.cfg.URL.Host)})
+					lblSet[HostLabel] = c.cfg.URL.Host
+					// also set client name since if we have multiple promtail clients configured we will run into a
+					// duplicate metric collected with same labels error when trying to hit the /metrics endpoint
+					lblSet[ClientLabel] = c.name
```

Comment:
id or config id would have work better IMO.

Reply:
for the configuration field name?